app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/item/MessageItemComponents.kt

```diff
@@ -274,14 +280,37 @@ internal fun MessageDecryptionFailure(
 
 @Composable
 internal fun Modifier.customizeMessageBackground(
+    defaultBackgroundColor: Color,
     sendingFailed: Boolean,
-    receivingFailed: Boolean
-) = run {
-    if (sendingFailed || receivingFailed) {
-        background(MaterialTheme.wireColorScheme.messageErrorBackgroundColor)
-    } else {
-        this
-    }
+    receivingFailed: Boolean,
+    isDeleted: Boolean,
+    isSelectedMessage: Boolean,
+    selfDeletionTimerState: SelfDeletionTimerHelper.SelfDeletionTimerState,
+    colorAnimation: Color
+): Modifier {
+
+    return this
+        .then(if (isSelectedMessage) drawBehind { drawRect(colorAnimation) } else this)
+        .then(
+            when {
+                sendingFailed || receivingFailed -> {
+                    background(MaterialTheme.wireColorScheme.messageErrorBackgroundColor)
+                }
+
+                selfDeletionTimerState is SelfDeletionTimerHelper.SelfDeletionTimerState.Expirable && !isDeleted -> {
+                    val color by animateColorAsState(
+                        targetValue = colorsScheme().primaryVariant.copy(alpha = selfDeletionTimerState.alphaBackgroundColor()),
+                        animationSpec = tween(),
+                        label = "message background color"
+                    )
+                    background(color)
```

Comment:
`drawBehind { drawRect(colorAnimation) }` is better for animations than `background(color)`